templates/cluster-template-external-loadbalancer.yaml

```diff
@@ -448,49 +665,58 @@ data:
               value: /csi/csi.sock
             - name: DRIVER_REG_SOCK_PATH
               value: /var/lib/kubelet/plugins/csi.vsphere.vmware.com/csi.sock
-            image: quay.io/k8scsi/csi-node-driver-registrar:v2.0.1
-            lifecycle:
-              preStop:
-                exec:
-                  command:
-                  - /bin/sh
-                  - -c
-                  - rm -rf /registration/csi.vsphere.vmware.com-reg.sock /csi/csi.sock
+            image: registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.8.0
+            livenessProbe:
+              exec:
+                command:
+                - /csi-node-driver-registrar
+                - --kubelet-registration-path=/var/lib/kubelet/plugins/csi.vsphere.vmware.com/csi.sock
+                - --mode=kubelet-registration-probe
+              initialDelaySeconds: 3
             name: node-driver-registrar
             resources: {}
-            securityContext:
-              privileged: true
             volumeMounts:
             - mountPath: /csi
               name: plugin-dir
             - mountPath: /registration
               name: registration-dir
-          - env:
+          - args:
+            - --fss-name=internal-feature-states.csi.vsphere.vmware.com
+            - --fss-namespace=$(CSI_NAMESPACE)
+            env:
+            - name: NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
             - name: CSI_ENDPOINT
               value: unix:///csi/csi.sock
+            - name: MAX_VOLUMES_PER_NODE
+              value: "59"
             - name: X_CSI_MODE
               value: node
             - name: X_CSI_SPEC_REQ_VALIDATION
               value: "false"
-            - name: VSPHERE_CSI_CONFIG
-              value: /etc/cloud/csi-vsphere.conf
+            - name: X_CSI_SPEC_DISABLE_LEN_CHECK
+              value: "true"
             - name: LOGGER_LEVEL
               value: PRODUCTION
-            - name: X_CSI_LOG_LEVEL
-              value: INFO
-            - name: NODE_NAME
+            - name: GODEBUG
+              value: x509sha1=1
```

Comment:
Can we add a comment why this one is needed? Or is this simply copy/paste?

Reply:
It's copied from the CSI manifest.
The flag is used to enable the deprecated SHA1 certificate support in some enterprise environment for on-prem Kubernetes installations.
https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/616f90f7e38a24fceda3673819abafd20da08859/manifests/vanilla/vsphere-csi-driver.yaml#L367C1-L368C32